### DISCUSSION TITLE
FYI, closing window problem when using fzf-lua with LazyVim

### DISCUSSION THREAD

**Discussion by m-fonseca** [LazyVim/LazyVim#1144] [General]
For those perhaps having this problem, the default lazyvim <esc><esc> mapping for the terminal interferes with the fzf-lua finder.  It
1. causes a long delay to exit the fuzzy finder window (whatever your `timeoutlen` is set to)
2. The long delay will lead the impatient (me) hitting escape more then once, which causes the window to not close (and you have to close it with another method like <kbd>ctrl-c</kbd><kbd>w</kbd> or <kbd>ctrl-d</kbd> etc.

See ibhagwan/fzf-lua#821 for more info

**Comment by willian** [comment]
It does not matter how many times I hit `<esc>`, the fzf-lua window never closes. To close it, I had to:
1. hit `<esc`>
2. type `:q`
3. hit `<cr>`

This is annoying; with Telescope, I had to simply hit `<esc>` whenever I wanted to close the window without selecting any option.

@m-fonseca, have you found a way to fix it?
@folke, or did you? :)

Thanks,

**Reply by dpetka2001** [reply]
[Screencast_2025-01-02-19-19-39.webm](https://github.com/user-attachments/assets/db9fdae4-bd27-4d2f-8a46-f2600b235405)

Fresh LazyVim installation, showing the mappings pressed. The fzf-lua window closes immediately from my testing.

**Comment by ibhagwan** [comment]
@m-fonseca, I could add a local buffer mapping that maps `esc` to itself with `nowait` but that would override the user settings (some users actually want the “normal” term mode with fzf-lua):

One possible solution (perhaps can be added to LazyVim) is adding an autocmd with `nowait` for fzf-lua:
```lua
  vim.api.nvim_create_autocmd("FileType", {
    group = vim.api.nvim_create_augroup("FzfLuaEsc", { clear = true }),
    pattern = "fzf",
    callback = function(e)
      vim.keymap.set("t", "<Esc>", "<Esc>", { buffer = e.buf, silent = true, nowait = true })
    end,
  })
```

> It does not matter how many times I hit <esc>, the fzf-lua window never closes. To close it, I had to:

@willian, this seems like a bug where your interface was opened in Normal mode instead, please update the plugin (fixed an issue related to this yesterday) and if this still happens please open an issue in the fzf-lua repo.

**Reply by dpetka2001** [reply]
@ibhagwan That mapping which is also mentioned in https://github.com/ibhagwan/fzf-lua/issues/821#issuecomment-1636826845 is no longer available in LazyVim. There's only a similar mapping in `Snacks.terminal` when the terminal opens, but that shouldn't have any effect in FzfLua. In the video that I recorded above with a fresh LazyVim installation, pressing `<esc>` instantly closes the FzfLua window. When no FzfLua window is opened, then `tmap <esc>` returns `no mapping found`. With FzfLua window open it returns `t  <Esc>       *@<Esc>`.

**Reply by ibhagwan** [reply]
> @ibhagwan That mapping which is also mentioned in [ibhagwan/fzf-lua#821 (comment)](https://github.com/ibhagwan/fzf-lua/issues/821#issuecomment-1636826845) is no longer available in LazyVim. There's only a similar mapping in `Snacks.terminal` when the terminal opens, but that shouldn't have any effect in FzfLua. In the video that I recorded above with a fresh LazyVim installation, pressing `<esc>` instantly closes the FzfLua window. When no FzfLua window is opened, then `tmap <esc>` returns `no mapping found`. With FzfLua window open it returns `t <Esc> *@<Esc>`.

Then that would be a user config issue as the mapping comes from their config.

**Reply by willian** [reply]
Guys, you're correct, it was my own config messing up. I just fixed: https://github.com/willian/dotfiles/commit/377986889e5b59ea4a1245d10c2622157bdb67a6

**Reply by m-fonseca** [reply]
FWIW, I haven't seen this problem in quite some time.  I don't know when but it seems to have been resolved.
